tests/nosetests/pyanaconda_tests/module_snapshot_test.py

```diff
@@ -0,0 +1,133 @@
+#
+# Copyright (C) 2019  Red Hat, Inc.
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# the GNU General Public License v.2, or (at your option) any later version.
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY expressed or implied, including the implied warranties of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
+# Public License for more details.  You should have received a copy of the
+# GNU General Public License along with this program; if not, write to the
+# Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+# 02110-1301, USA.  Any Red Hat trademarks that are incorporated in the
+# source code or documentation are not subject to the GNU General Public
+# License and may only be used or replicated with the express permission of
+# Red Hat, Inc.
+#
+# Red Hat Author(s): Vendula Poncova <vponcova@redhat.com>
+#
+import unittest
+from unittest.mock import patch, Mock
+
+from pykickstart.constants import SNAPSHOT_WHEN_POST_INSTALL, SNAPSHOT_WHEN_PRE_INSTALL
+from pykickstart.errors import KickstartParseError
+
+from pyanaconda.modules.common.errors.storage import UnavailableStorageError
+from pyanaconda.modules.common.task import TaskInterface
+from pyanaconda.modules.storage.snapshot import SnapshotModule
+from pyanaconda.modules.storage.snapshot.create import SnapshotCreateTask
+from pyanaconda.modules.storage.snapshot.device import get_snapshot_device
+from pyanaconda.modules.storage.snapshot.snapshot_interface import SnapshotInterface
+from pyanaconda.modules.storage.snapshot.validate import SnapshotValidateTask
+
+
+class SnapshotInterfaceTestCase(unittest.TestCase):
+    """Test DBus interface of the Snapshot module."""
+
+    def setUp(self):
+        """Set up the module."""
+        self.module = SnapshotModule()
+        self.interface = SnapshotInterface(self.module)
+
+    def is_requested_test(self):
+        """Test IsRequested."""
+        self.assertEqual(self.interface.IsRequested(SNAPSHOT_WHEN_PRE_INSTALL), False)
+        self.assertEqual(self.interface.IsRequested(SNAPSHOT_WHEN_POST_INSTALL), False)
+
+        self.module._requests = [Mock(when=SNAPSHOT_WHEN_PRE_INSTALL)]
+        self.assertEqual(self.interface.IsRequested(SNAPSHOT_WHEN_PRE_INSTALL), True)
+        self.assertEqual(self.interface.IsRequested(SNAPSHOT_WHEN_POST_INSTALL), False)
+
+        self.module._requests = [Mock(when=SNAPSHOT_WHEN_POST_INSTALL)]
+        self.assertEqual(self.interface.IsRequested(SNAPSHOT_WHEN_PRE_INSTALL), False)
+        self.assertEqual(self.interface.IsRequested(SNAPSHOT_WHEN_POST_INSTALL), True)
```

Comment:
You may add test when both are set. The result should be the last wins if I'm not mistaken.

Reply:
By the both I mean `SNAPSHOT_WHEN_(PRE|POST)_INSTALL` :).

Reply:
Fixed.